runtime/docker/container.go

```diff
@@ -116,6 +117,13 @@ func (c *client) RunContainer(ctx context.Context, ctn *pipeline.Container, b *p
 	c.ctnConf = ctnConfig(ctn)
 	c.netConf = netConfig(b.ID, ctn.Name)
 
+	// check if the image is allowed to run privileged
+	for _, image := range c.privilegedImages {
+		if strings.HasPrefix(ctn.Image, image) {
```

Comment:
I think this should be further updated to hopefully prevent someone from maliciously creating a image that utilizes the same prefix.

```suggestion
		if strings.HasPrefix(ctn.Image, fmt.Sprintf("%s:", image) || ctn.Image == image
```

Reply:
Took a slightly different approach. Then suggested, I opened up the list of privileged images to leverage Docker's reference match functionality.
https://pkg.go.dev/github.com/docker/distribution/reference#FamiliarMatch

I think this will allow us to stay closer to Docker's preferred way to match image types which should allow the administrator to compose a list of acceptable images to be used for a privileged image.

Reply:
> Took a slightly different approach. Then suggested, I opened up the list of privileged images to leverage Docker's reference match functionality.
> https://pkg.go.dev/github.com/docker/distribution/reference#FamiliarMatch
> 
> I think this will allow us to stay closer to Docker's preferred way to match image types which should allow the administrator to compose a list of acceptable images to be used for a privileged image.

Sounds good! Can you add some more tests that validate this updated function will resolve my concerns? e.g. the pattern of `docker.company.com/foo` does not match the image of `docker.company.com/foo/bar`.

Reply:
Sure! I'll see if I can add that one in there!

Reply:
This commit I believe highlights the concern:
https://github.com/go-vela/pkg-runtime/pull/62/commits/363491db8c42a42179721eb709cb08164e0adfbc

So, in the test, we are expecting admin to have to allow listed: `docker.company.com/foo` with any tag available.

If a user tries to create and use `docker.company.com/foo/bar` that image on all tags would be blocked from getting the socket mount.